cmd/mailbox-controller/controller.go

```diff
@@ -187,6 +204,20 @@ func (ctl *mbCtl) sync1(ctx context.Context, ref any) {
 
 func (ctl *mbCtl) sync(ctx context.Context, refany any) bool {
 	logger := klog.FromContext(ctx)
+	stName, ok := refany.(refSyncTarget)
+	if ok {
+		st, err := ctl.synctargetLister.Get(string(stName))
+		if err != nil {
+			logger.Error(err, "Failed to fetch SyncTarget from local cache", "stName", stName)
+			return false
+		}
+		mbwsName, err := ctl.mbwsNameOfSynctarget(st)
+		if err != nil {
+			return true
+		}
+		ctl.queue.Add(mbwsName)
+		return false
```

Comment:
Why go to all this trouble rather than just falling through to the mbwsName handling right below?